app/controllers/Application.scala

```diff
@@ -4,14 +4,35 @@ import play.api._
 import play.api.mvc._
 import services._
 
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+
 object Application extends Controller {
 
   def index = Action {
     Ok(views.html.index("Your new application is ready."))
   }
 
-  def search(q: Option[String]) = Action {
+  def search(q: Option[String]) = Action.async {
     val results: Option[SearchResults] = q.map(SearchService.get)
```

Comment:
You could also wrap this in a future, to have the whole Action return as fast as possible.